File: sample/use_cocoapods/objectivec/GNAdDFPRewardAdapterSample/GNAdDFPRewardMediationAdapterSample/ViewController.m
//
//  ViewController.m
//  GNAdDFPRewardMediationAdapterSample
//

#import "ViewController.h"

@interface ViewController ()

/// Is an ad being loaded.
@property(nonatomic, assign, getter=isRewardBasedVideoRequestLoading) BOOL rewardBasedVideoRequestLoading;

/// The countdown timer.
@property(nonatomic, strong) NSTimer *timer;
/// The game counter.
@property(nonatomic, assign) NSInteger counter;

/// Number of coins the user has earned.
@property(nonatomic, assign) NSInteger coinCount;
@property (weak, nonatomic) IBOutlet UITextField *dfpAdUnitIdLabel;

@end



@implementation ViewController



- (void)viewDidLoad {
    [super viewDidLoad];
    
    [GADRewardBasedVideoAd sharedInstance].delegate = self;
    
    self.showVideoButton.hidden = YES;
}


- (IBAction)showVideo:(id)sender {
    if ([[GADRewardBasedVideoAd sharedInstance] isReady]) {
        [[GADRewardBasedVideoAd sharedInstance] presentFromRootViewController:self];
    }
}

- (IBAction)playAgain:(id)sender {
    
    DFPRequest *request = [DFPRequest request];
    
    [[GADRewardBasedVideoAd sharedInstance] loadRequest:request withAdUnitID:_dfpAdUnitIdLabel.text];
    
    self.showVideoButton.hidden = YES;
}

#pragma mark: GADRewardBasedVideoAdDelegate

- (void)rewardBasedVideoAd:(GADRewardBasedVideoAd *)rewardBasedVideoAd didRewardUserWithReward:(GADAdReward *)reward {
    
    NSString *rewardMessage =
    [NSString stringWithFormat:@"Reward received with currency %@ , amount %lf",
     reward.type,
     [reward.amount doubleValue]];
    NSLog(@"didRewardUserWithReward %@", rewardMessage);
}

- (void)rewardBasedVideoAd:(GADRewardBasedVideoAd *)rewardBasedVideoAd didFailToLoadWithError:(NSError *)error {
    NSLog(@"didFailToLoadWithError error = %@", error);
}

- (void)rewardBasedVideoAdDidReceiveAd:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    
    NSLog(@"rewardBasedVideoAdDidReceiveAd");
    
    self.showVideoButton.hidden = NO;
}

/// Tells the delegate that the reward based video ad opened.
- (void)rewardBasedVideoAdDidOpen:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"rewardBasedVideoAdDidOpen");
}

/// Tells the delegate that the reward based video ad started playing.
- (void)rewardBasedVideoAdDidStartPlaying:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"rewardBasedVideoAdDidStartPlaying");
}

/// Tells the delegate that the reward based video ad completed playing.
- (void)rewardBasedVideoAdDidCompletePlaying:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"rewardBasedVideoAdDidCompletePlaying");
}

/// Tells the delegate that the reward based video ad closed.
- (void)rewardBasedVideoAdDidClose:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"rewardBasedVideoAdDidClose");
    
    self.showVideoButton.hidden = YES;
}

/// Tells the delegate that the reward based video ad will leave the application.
- (void)rewardBasedVideoAdWillLeaveApplication:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"rewardBasedVideoAdWillLeaveApplication");
}


@end

File: sample/use_cocoapods/swift/GNAdDFPRewardAdapterSample/GNAdDFPRewardAdapterSample/ViewController.swift
//
//  ViewController.swift
//  GNAdDFPRewardedAdapterSample
//

import GoogleMobileAds
import UIKit

class ViewController: UIViewController {

    /// The reward-based video ad.
    var rewardBasedVideo: GADRewardBasedVideoAd?
    
    @IBOutlet weak var unitIdTextField: UITextField!
    @IBOutlet weak var loadButton: UIButton!
    @IBOutlet weak var showButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        unitIdTextField.delegate = self
        rewardBasedVideo = GADRewardBasedVideoAd.sharedInstance()
        rewardBasedVideo?.delegate = self
        loadButton.isHidden = false
        showButton.isHidden = true
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    // MARK: Button actions
    @IBAction func touchLoadButton(_ sender: Any) {
        if rewardBasedVideo?.isReady == false {
            rewardBasedVideo?.load(DFPRequest(),
                                   withAdUnitID: unitIdTextField.text!)
            loadButton.isHidden = true;
        }
    }
    
    @IBAction func touchShowButton(_ sender: Any) {
        if rewardBasedVideo?.isReady == true {
            rewardBasedVideo?.present(fromRootViewController: self)
            loadButton.isHidden = false
            showButton.isHidden = true
        }
    }
}
// MARK: GADRewardBasedVideoAdDelegate implementation
extension ViewController : GADRewardBasedVideoAdDelegate {
    
    func rewardBasedVideoAd(_ rewardBasedVideoAd: GADRewardBasedVideoAd,
                            didFailToLoadWithError error: Error) {
        print("Reward based video ad failed to load: \(error.localizedDescription)")
        loadButton.isHidden = false
    }
    
    func rewardBasedVideoAdDidReceive(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print("Reward based video ad is received.")
        showButton.isHidden = false
    }
    
    func rewardBasedVideoAdDidOpen(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print("Opened reward based video ad.")
    }
    
    func rewardBasedVideoAdDidStartPlaying(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print("Reward based video ad started playing. : " + (rewardBasedVideoAd.adNetworkClassName?.description)!)
    }
    
    func rewardBasedVideoAdDidClose(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print("Reward based video ad is closed.")
    }
    
    func rewardBasedVideoAdWillLeaveApplication(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print("Reward based video ad will leave application.")
    }
    
    func rewardBasedVideoAd(_ rewardBasedVideoAd: GADRewardBasedVideoAd,
                            didRewardUserWith reward: GADAdReward) {
        print("Reward received with currency: \(reward.type), amount \(reward.amount).")
    }
}
// MARK: UITextFieldDelegate implementation
extension ViewController : UITextFieldDelegate {
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        // Hide the keyboard when press return key in a UITextField
        textField.resignFirstResponder()
        return true;
    }
}

